### PR TITLE
Update pollution sensor data every 10 minutes

### PR DESCRIPTION
**Pull request by Yqiyuan**
# Description

Update pollution sensor data every 10 minutes

# Test Plan

Please explain how this change will be tested.

- [x] Unit test
- [x] Integration test (including unit tests with supertest)
- [x] Manual test (Please explain below for the test steps)
- [ ] Not Applicable

# Task Link

Please add related jira task links

### CODE REVIEW

**Review by bettsongy — APPROVED** [review]

File: server/src/utils/PollutionDataUpdater.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  private constructor(interval = 600000, updateMaxNumber = 6) {
+    this.interval = interval
+    this.updateNumber = updateMaxNumber
+    this.updateMaxNumber = updateMaxNumber
+  }
+
+  public static get Instance() {
+    return this._instance || (this._instance = new PollutionDataUpdater())
+  }
```
Comment (jennyzeng) [inline server/src/utils/PollutionDataUpdater.ts:23]: is there any case that you couldn't just create an instance in this file and export the instance as default? That would also guarantee to be singleton

File: server/src/utils/PollutionDataUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        await sensor.save()
+        await new Promise((resolve) => setTimeout(resolve, 2000))
+      }
+    }
+
+    this.updateNumber -= 1
+
+    const users = UserConnections.getConnectedUsers()
+    for (const user of users) {
+      const connection = UserConnections.getUserConnection(user)
```
Comment (jennyzeng) [inline server/src/utils/PollutionDataUpdater.ts:73]: nit: you can just add a function in `UserConnections` that return you a list of connections that you can use to emit. So you don't need to get a list of keys and then get connection by key

File: server/src/utils/PollutionDataUpdater.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    // Update pollution data
+    const pollutionSensors = await MapElement.find({
+      type: 'pollution',
+    })
+    for (const sensor of pollutionSensors) {
+      if (sensor.point?.coordinates) {
+        const position = sensor.point.coordinates
+        const pollutionData = await getPollutionData(position[1], position[0])
+        if (sensor.pollutionData) {
```
Comment (jennyzeng) [inline server/src/utils/PollutionDataUpdater.ts:60]: how will you catch exceptions? in the case exceptions happen, will this stop and not run?

**Review by jennyzeng — CHANGES_REQUESTED** [review]

File: server/src/utils/PollutionDataUpdater.ts (hunk 1 above)
Comment (Yqiyuan) [reply server/src/utils/PollutionDataUpdater.ts]: Good suggestion. I exported the instance.

File: server/src/utils/PollutionDataUpdater.ts (hunk 2 above)
Comment (Yqiyuan) [reply server/src/utils/PollutionDataUpdater.ts]: Added

File: server/src/utils/PollutionDataUpdater.ts (hunk 3 above)
Comment (Yqiyuan) [reply server/src/utils/PollutionDataUpdater.ts:59]: Thank you. I catch the error now

File: server/src/utils/PollutionDataUpdater.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import MapElement from '../models/MapElement'
+import { getPollutionData } from './pollution'
+import UserConnections from './UserConnections'
+
+class PollutionDataUpdater {
+  static _instance: PollutionDataUpdater
```
Comment (jennyzeng) [inline server/src/utils/PollutionDataUpdater.ts:6]: Remove this?

**Review by jennyzeng — APPROVED** [review]

File: server/src/utils/PollutionDataUpdater.ts (hunk 4 above)
Comment (Yqiyuan) [reply server/src/utils/PollutionDataUpdater.ts]: Thanks. Removed.

**Pull request merged**